app/src/main/java/com/nononsenseapps/feeder/archmodel/SettingsStore.kt

```diff
@@ -476,6 +476,7 @@ class SettingsStore(
             .putString(PREF_OPENAI_URL, value.baseUrl)
             .putString(PREF_OPENAI_AZURE_VERSION, value.azureApiVersion)
             .putString(PREF_OPENAI_AZURE_DEPLOYMENT_ID, value.azureDeploymentId)
+            .putInt(PREF_OPENAI_REQUEST_TIMEOUT_SECONDS, value.timeoutSeconds.coerceIn(30, 600))
```

Comment:
```suggestion
            .putInt(PREF_OPENAI_REQUEST_TIMEOUT_SECONDS, value.timeoutSeconds)
```

An incorrect value should never be allowed to be put into the datastructure in the first place. And this coercion doesn't fix possible issues present in the `OpenAISettings` object

Reply:
I do agree with you but I can't find any other places where I can call coerceIn and fix the `OpenAISettings` object properly beforehand.

If calling it as part of onValueChange, it will be called on every keystroke, and that will interfere user input, and you also don't like calling it as part of onFocusChange which I did initially.

Reply:
Just remove the coercion from here and it's fine